components/post_view/post_info/index.js

```diff
@@ -4,15 +4,19 @@
 import {connect} from 'react-redux';
 import {bindActionCreators} from 'redux';
 import {addReaction, removePost} from 'mattermost-redux/actions/posts';
+import {getChannel} from 'mattermost-redux/selectors/entities/channels';
 import {get, getBool} from 'mattermost-redux/selectors/entities/preferences';
 
 import {Preferences} from 'utils/constants.jsx';
 
 import PostInfo from './post_info.jsx';
 
 function mapStateToProps(state, ownProps) {
+    const channel = getChannel(state, {id: ownProps.post.channel_id}) || {};
+    const teamId = channel.team_id;
```

Comment:
I think this will fail if channel is empty object like in the case of `|| {}` above.

Reply:
The approach is, getChannel returns something or "nothing" if nothing, then channel will be an empty object `{}`, if you ask an empty object for a property, for example, team_id, and that property doesn't exist, it returns "undefined", but not fail. Is my expected behavior, because anyway, `teamId` is a Required field, and if I have the post, I must have the channel, and if I have the channel, I must have the `team_id`.

Reply:
You're definitely correct. What I should have said is, it will fail on `teamId...isRequired` because of probable `undefined` value.
Not blocking, just thought to have protection in place.